packages/flutter/lib/src/painting/decoration_image.dart

```diff
@@ -642,9 +643,23 @@ void paintImage({
     if (repeat == ImageRepeat.noRepeat) {
       canvas.drawImageRect(image, sourceRect, destinationRect, paint);
     } else {
-      for (final Rect tileRect in _generateImageTileRects(rect, destinationRect, repeat)) {
-        canvas.drawImageRect(image, sourceRect, tileRect, paint);
-      }
+      final TileMode tmx = (repeat == ImageRepeat.repeatX || repeat == ImageRepeat.repeat)
+        ? TileMode.repeated
+        : TileMode.clamp;
+      final TileMode tmy = repeat == ImageRepeat.repeatY || repeat == ImageRepeat.repeat
```

Comment:
nit: style this the same as in line 646 for consistency? i.e. add parentheses here or remove them in line 646?

Reply:
Done